test/conversions.jl

```diff
@@ -114,29 +182,232 @@ end
+    @test promote_type(TransparentColor{RGB{N0f8},N0f8,4}, RGB{Float64}) === ARGB{Float64}
+    @test promote_type(AlphaColor, RGB{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(AlphaColor{RGB{N0f8}}, RGB{Float64}) === ARGB
+    @test promote_type(AlphaColor{RGB{N0f8},N0f8}, RGB{Float64}) === ARGB{Float64}
+    @test promote_type(AlphaColor{RGB{N0f8},N0f8,4}, RGB{Float64}) === ARGB{Float64}
+    @test promote_type(ColorAlpha, RGB{Float64}) === ColorAlpha{C,T} where {T, C<:Color{T}}
+    @test promote_type(ColorAlpha{RGB{N0f8}}, RGB{Float64}) === RGBA
+    @test promote_type(ColorAlpha{RGB{N0f8},N0f8}, RGB{Float64}) === RGBA{Float64}
+    @test promote_type(ColorAlpha{RGB{N0f8},N0f8,4}, RGB{Float64}) === RGBA{Float64}
+
+    @test promote_type(Colorant, ARGB{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(Colorant{N0f8}, ARGB{Float64}) == AlphaColor{C,Float64} where C<:Color{Float64}
+    @test_broken promote_type(Colorant{Float32,4}, ARGB{Float64}) == AlphaColor{C,Float64,4} where C<:Color3{Float64}
+    @test promote_type(Color, ARGB{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(AbstractRGB, ARGB{Float64}) === ARGB
+    @test promote_type(AbstractRGB{N0f8}, ARGB{Float64}) === ARGB{Float64}
+    @test promote_type(TransparentColor, ARGB{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(AlphaColor, ARGB{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(ColorAlpha, ARGB{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+
+    @test promote_type(Colorant, RGBA{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(Colorant{N0f8}, RGBA{Float64}) == AlphaColor{C,Float64} where C<:Color{Float64}
+    @test promote_type(Colorant{Float32,3}, RGBA{Float64}) == AlphaColor{C,Float64,4} where C<:Color3{Float64}
+    @test promote_type(Color, RGBA{Float64}) == ColorAlpha{C,T} where {T, C<:Color{T}}
+    @test promote_type(AbstractRGB, RGBA{Float64}) === RGBA
+    @test promote_type(AbstractRGB{N0f8}, RGBA{Float64}) === RGBA{Float64}
+    @test promote_type(TransparentColor, RGBA{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(AlphaColor, RGBA{Float64}) == AlphaColor{C,T} where {T, C<:Color{T}}
+    @test promote_type(ColorAlpha, RGBA{Float64}) == ColorAlpha{C,T} where {T, C<:Color{T}}
 end
 
+
 @testset "rgb conversions with abstract types" begin
     c = RGB(1, 0.6, 0)
     @test convert(Colorant, c) === RGB{Float64}(1, 0.6, 0)
     @test convert(Colorant{N0f8}, c) === RGB{N0f8}(1, 0.6, 0)
     @test convert(Colorant{Float32,3}, c) === RGB{Float32}(1, 0.6, 0)
     @test convert(Color, c) === RGB{Float64}(1, 0.6, 0)
     @test convert(Color{N0f8}, c) === RGB{N0f8}(1, 0.6, 0)
-    @test convert(Color{Float32}, c) === RGB{Float32}(1, 0.6, 0)
+    @test convert(Color{Float32,3}, c) === RGB{Float32}(1, 0.6, 0)
```

Comment:
Does it have trouble without the 3?

Reply:
There is no trouble. However, `convert(Color{N0f8}, c)` and `convert(Color{Float32}, c)` are essentially the same case.